src/core/server/status/legacy_status.ts

```diff
@@ -0,0 +1,152 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { pick } from 'lodash';
+import { i18n } from '@kbn/i18n';
+import { ServiceStatusLevels, ServiceStatus, CoreStatus } from './types';
+import { deepFreeze } from '../../utils';
+import { PluginName } from '../plugins';
+
+interface Deps {
+  overall: ServiceStatus;
+  core: CoreStatus;
+  plugins: Record<PluginName, ServiceStatus>;
+  versionWithoutSnapshot: string;
+}
+
+export interface LegacyStatusInfo {
+  overall: {
+    state: LegacyStatusState;
+    title: string;
+    nickname: string;
+    uiColor: LegacyStatusUiColor;
+    /** ISO-8601 date string w/o timezone */
+    since: string;
+    icon?: string;
+  };
+  statuses: StatusComponentHttp[];
+}
+
+export const calculateLegacyStatus = ({
+  core,
+  overall,
+  plugins,
+  versionWithoutSnapshot,
+}: Deps): LegacyStatusInfo => {
+  const since = new Date().toISOString();
+  const overallLegacy: LegacyStatusInfo['overall'] = {
+    since,
+    ...pick(STATUS_LEVEL_LEGACY_ATTRS[overall.level.toString()], [
+      'state',
+      'title',
+      'nickname',
+      'icon',
+      'uiColor',
+    ]),
+  };
+  const coreStatuses = Object.entries(core).map(([serviceName, s]) =>
+    serviceStatusToHttpComponent(`core:${serviceName}@${versionWithoutSnapshot}`, s, since)
+  );
+  const pluginStatuses = Object.entries(plugins).map(([pluginName, s]) =>
+    serviceStatusToHttpComponent(`plugin:${pluginName}@${versionWithoutSnapshot}`, s, since)
+  );
+
+  const componentStatuses: StatusComponentHttp[] = [...coreStatuses, ...pluginStatuses];
```

Comment:
I included them only because `elasticsearch` was previously a plugin and this ensures that that is still exposed. There is a side-benefit that the existing Status Page UI will show these statuses without any modifications.